### ISSUE TITLE
Trust-proxy, prebuilt-deps, docker container

### ISSUE THREAD

**Issue by RnbWd** [rlidwka/sinopia#191]
**TL:DR;**

* node-pre-gyp on `crypt3` and `fs-ext` == faster builds and less headaches
* `app.use('trust-proxy')` helps when using in front of an nginx server running ssl/tls.
* `core-js` vs. `es-shim`? (does it even matter)

I'm hosting some docker containers: [rnbwd/sinopia](https://registry.hub.docker.com/u/rnbwd/sinopia/) and [dympl/sinopia-registry](https://registry.hub.docker.com/u/dympl/sinopia-registry/). The first container clones this repo (at v1.0.0-beta.2) inside of the container itself, while the second repo is a fork of sinopia  with quite a few modifications that may be of interest..

1) It uses [crypt3-prebuilt](https://www.npmjs.com/package/crypt3-prebuilt) and [fs-ext-prebuilt](https://www.npmjs.com/package/fs-ext-prebuilt) as deps for [sinopia-htpasswd-prebuilt](https://github.com/codingalchemy/sinopia-htpasswd-prebuilt):
 
Using pre-gyp, I was able to generate crypt3 and fs-ext builds for x64 linux / darwin that cut down the npm install time substantially and may reduce the need for using npm-shrinkwrap because crypt3 and fs-ext don't build properly without the shrinkwrap. I could use some help compiling distros for more OS's (windows???). It could be used as a plugin on it's own too independent of docker.

2) `app.enable('trust-proxy)`

I use  [nginx-proxy](https://registry.hub.docker.com/u/jwilder/nginx-proxy/) (or my own [docker forks](https://registry.hub.docker.com/u/rnbwd/spdy-proxy/)) for hosting sinopia, and I've always had to change the `url-prefix` in config.yaml to get it working in front of the https://. Enabling `trust-proxy` allowed me to use sinopia without changing any configs, and it seems to work with multiple uri's (which is kinda cool). `env.VIRTUAL_HOST` is passed by docker into node, so I know that I'm using the proxy before enabling it via `if (process.env.VIRTUAL_HOST) app.enable('trust-proxy)`. I'm not sure what (if any) consequences there would be to enable `trust-proxy`, and that's something VERY important to know. 

3) replaced `es6-shim` with `core-js`

`6to5` uses `core-js` to shim `es6`. Apparently it has more features and better perf. I dunno, I just use `6to5` in all my projects, so I switched it, and all the tests work fine. So, maybe something to consider? (might not make a difference).

4) Testing and Building Frontend

A few (minor) changes were necessary to get the tests working, and I recompiled the front-end via gulp (minified with uglify / csso), but end product is the same. I'm interested in contributing to the UI at some point when I'm not so busy.

**Comment by rlidwka** [comment]
Thanks for you work.

1\. I don't know anything about pre-built packages. Maybe it's worth mentioning/discussing in [fs-ext](https://github.com/baudehlo/node-fs-ext) issue list, since many people use that package.

2\. About `trust-proxy`. Only consequence would be that if you have `trust-proxy` enabled and don't use proxy (or misconfigure it), an attacker would be able to mangle his ip address in logs. We don't use any kind of ip-based access control, so it's not extremely important. I wouldn't turn it on by default though.

3\. I don't care about `es6-shim` very much. It's used in like 2 places, so we can even remove it completely. I'll check `core-js` though.

**Comment by RnbWd** [comment]
1) I don't know much about pre-built packages either.. but I realized that there was a `nan` dependency issue between `crypt-3` and `fs-ext` in terms of which versions of `nan` they used and how explicit they were in their package.json. 

2) This isn't really an issue so I'll close.

3) I love using sinopia, thank you so much for building it!!
